### PR TITLE
develop

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
tests are failed

File: services/cinema_hall.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+def get_cinema_halls() -> CinemaHall:
+    return CinemaHall.objects.all()
+
+
+def create_cinema_hall(
+        hall_name: str,
+        hall_rows: int,
+        hall_seats_in_row: int
+) -> None:
```
Comment (ArsenPidhoretskyi) [inline services/cinema_hall.py:12]: It does not return None
It returns CinemaHall

File: services/cinema_hall.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from db.models import CinemaHall
+
+
+def get_cinema_halls() -> CinemaHall:
```
Comment (ArsenPidhoretskyi) [inline services/cinema_hall.py:4]: .all does not return one object

File: services/movie.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: list[int] = None,
```
Comment (ArsenPidhoretskyi) [inline services/movie.py:5]: Use Optional[list[int]] because None as default provided
Fix in all places with all types

File: services/movie.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> Movie:
```
Comment (ArsenPidhoretskyi) [inline services/movie.py:7]: Wrong return type annotations

File: services/movie.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(id=movie_id)
+
+
+def create_movie(
+        movie_title: str,
+        movie_description: str,
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> None:
```
Comment (ArsenPidhoretskyi) [inline services/movie.py:28]: Wrong return type annotations

File: services/movie_session.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+    return new_movie_session
+
+
+def get_movies_sessions(
+        session_date: datetime = None
+) -> MovieSession:
```
Comment (ArsenPidhoretskyi) [inline services/movie_session.py:22]: Wrong return type annotations

File: services/movie_session.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    movie_session = MovieSession.objects.all()
```
Comment (ArsenPidhoretskyi) [inline services/movie_session.py:23]: rename movie_session to movie_sessions

File: services/movie_session.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+            id=session_id
+        ).update(show_time=show_time)
+    if movie_id is not None:
+        MovieSession.objects.filter(
+            id=session_id
+        ).update(movie_id=movie_id)
+    if cinema_hall_id is not None:
+        MovieSession.objects.filter(
+            id=session_id
+        ).update(cinema_hall_id=cinema_hall_id)
```
Comment (ArsenPidhoretskyi) [inline services/movie_session.py:54]: No need to do 3 queries. You can do it by one query or use two queries:
1. Get object then set all new variables
2. Save object

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**